Src/main.c

```diff
@@ -103,13 +103,28 @@ int main(void)
 
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
   /* USER CODE END 2 */
+  //Initialize ADC and TIM
+  HAL_ADC_Init(&hadc);
+  HAL_TIM_PWM_Start(&htim16, TIM_CHANNEL_1);
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  //set servo PWM output relative to input voltage from potentiometer.
+  //potentiometer voltage from 0-3.3V, PWM modulation from 1-2ms on-time at 50Hz
+  double ADCconvert = 1023; //converting ADC to decimal
```

Comment:
Avoid declaring things as doubles unless they absolutely need to be - they take up unnecessary amounts of memory when a lower memory-intensive data type (i.e. uint16_t) could be used instead. 

I know you did this to fix the integer division issue a while back, so I'll leave you with this: try to cast the variable inside the operation. (If you're unsure, try to google how to cast things)

P.S. floats are preferable to doubles since they take up less memory :D

Reply:
Used unsigned ints as that's a little lower memory use. Not sure how large int sizes could be for potential uses. Also cast floats.